### DISCUSSION TITLE
Create_remote API

### DISCUSSION THREAD

**Discussion by gavishdewangan** [Altinity/clickhouse-backup#981] [Q&A] [answered]
Is there a a support for creating a backup in remote server through API, going through documentation I could find create_remote command but could not find any support through any API.
Can this API POST /backup/create be used for creating backup in the remote server?

**Comment by Slach** [comment, marked as answer]
look to `POST /backup/actions`

https://github.com/Altinity/clickhouse-backup/#post-backupactions

`curl -X POST -d '{"command":"create_remote test_backup"}' -s https://clickhouse-bakcup:7171/backup/actions`

**Comment by gavishdewangan** [comment]
I have a follow up question on this, as per the documentation this API accepts command as a parameter in the request body, do we have an option of sending remote_storage option and all other related parameter like host address to this api so that we can define where do we want to create backup?

**Reply by Slach** [reply]
`curl -X POST -d '{"command":"create_remote --env=REMOTE_STORAGE=s3 test_backup"}' -s https://clickhouse-bakcup:7171/backup/actions`

**Comment by gavishdewangan** [comment]
What about other parameters of s3
like 
s3:
  access_key: ""                   # S3_ACCESS_KEY
  secret_key: ""                   # S3_SECRET_KEY
  bucket: ""                       # S3_BUCKET
  endpoint: ""                     # S3_ENDPOINT
  region: us-east-1                # S3_REGION
  
and we can provide sftp as REMOTE_STORAGE right
and if we can then, same question for sftp also, how can we provide parameters of sftp
sftp:
  address: ""                  # SFTP_ADDRESS
  username: ""                 # SFTP_USERNAME
  password: ""                 # SFTP_PASSWORD
  port: 22                     # SFTP_PORT
  key: ""                      # SFTP_KEY
  path: ""                     # SFTP_PATH, `system.macros` values can be applied as {macro_name}
  concurrency: 1               # SFTP_CONCURRENCY
  compression_format: tar      # SFTP_COMPRESSION_FORMAT, allowed values tar, lz4, bzip2, gzip, sz, xz, brortli, zstd, `none` for upload data part folders as is
  compression_level: 1         # SFTP_COMPRESSION_LEVEL
  debug: false                 # SFTP_DEBUG

**Reply by Slach** [reply]
check https://github.com/Altinity/clickhouse-backup/?tab=readme-ov-file#explain-config-parameters
